pkg/network/setup/netstat.go

```diff
@@ -131,10 +131,19 @@ func (c *NetStat) UpdateStatus(vmi *v1.VirtualMachineInstance, domain *api.Domai
 		interfacesStatus = append([]v1.VirtualMachineInstanceNetworkInterface{*primaryInterfaceStatus}, interfacesStatus...)
 	}
 
-	for ifaceIndex, ifaceStatus := range interfacesStatus {
-		if _, exists := multusStatusNetworksByName[ifaceStatus.Name]; exists {
-			interfacesStatus[ifaceIndex].InfoSource = netvmispec.AddInfoSource(
-				ifaceStatus.InfoSource, netvmispec.InfoSourceMultusStatus)
+	indexedIfaceStatus := netvmispec.IndexInterfacesFromStatus(interfacesStatus,
+		func(ifaceStatus v1.VirtualMachineInstanceNetworkInterface) bool { return true })
+	for multusIfaceName := range multusStatusNetworksByName {
+		_, existInSpec := vmiInterfacesSpecByName[multusIfaceName]
+		_, existInNewStatus := indexedIfaceStatus[multusIfaceName]
+		if existInSpec && !existInNewStatus {
```

Comment:
I think we cannot depend on the status, we have been there before and it was found problematic.
It will be better to limit the dependency as much as possible, fetching information from the previous status with great care.

It will be enough to add an existing dependency in the virt-controller, to "leak" entries.

Going through the spec is a small attempt to avoid such a potential "leak" and reduce the dependency.

Reply:
You right, its not a must and works w/o it.
I thought it should be added because of this [test](https://github.com/kubevirt/kubevirt/blob/main/pkg/network/setup/netstat_test.go#L205), if fails if we don't check the iface exists in spec.

But I guess it's fine and the test should be changed because now we do expect to see such interfaces in the status.

Removed the checking against the spec and changed [unit-test](https://github.com/kubevirt/kubevirt/blob/main/pkg/network/setup/netstat_test.go#L205).

Reply:
I missed this comment https://github.com/kubevirt/kubevirt/pull/10063#discussion_r1258310610

Reply:
As Edy mentioned, initially it was an attempt to not rely on the previous status entirely.